cli/cmd/check.go

```diff
@@ -22,64 +25,79 @@ func newCheckCmd() *cobra.Command {
 			if err != nil {
 				return err
 			}
-			return cliCheck(kubeClient)
+			return cliCheck(kubeClient, timeout)
 		},
 		SilenceUsage: true,
 	}
 
+	cmd.Flags().UintVar(&timeout, "timeout", 60, "Time to wait before aborting in seconds")
 	return cmd
 }
 
-// check if marblerun control-plane deployments are ready to use
-func cliCheck(kubeClient kubernetes.Interface) error {
-	err := checkDeploymentStatus(kubeClient, "marble-injector", "marblerun")
-	if err != nil {
+// cliCheck if marblerun control-plane deployments are ready to use
+func cliCheck(kubeClient kubernetes.Interface, timeout uint) error {
+	if err := checkDeploymentStatus(kubeClient, "marble-injector", "marblerun", timeout); err != nil {
 		return err
 	}
 
-	err = checkDeploymentStatus(kubeClient, "marblerun-coordinator", "marblerun")
-	if err != nil {
+	if err := checkDeploymentStatus(kubeClient, "marblerun-coordinator", "marblerun", timeout); err != nil {
 		return err
 	}
 
-	// Add checks for other control plane deployments here
-	//
-	// err = checkDeploymentStatus(kubeClient, "some-control-plane-component", "marblerun")
-	//
-
 	return nil
 }
 
 // checkDeploymentStatus checks if a deployment is installed on the cluster
 // If it is, this function will wait until all replicas have the "available" status (ready for at least minReadySeconds)
 // Current status is continuously printed on screen
-func checkDeploymentStatus(kubeClient kubernetes.Interface, deploymentName string, namespace string) error {
+func checkDeploymentStatus(kubeClient kubernetes.Interface, deploymentName string, namespace string, timeout uint) error {
+	var tries uint
+	var podsReady string
+
+	// check if command is run from a terminal
+	isTTY := false
+	if fileInfo, _ := os.Stdout.Stat(); (fileInfo.Mode() & os.ModeCharDevice) != 0 {
+		isTTY = true
+	}
+
 	_, err := kubeClient.AppsV1().Deployments(namespace).Get(context.TODO(), deploymentName, metav1.GetOptions{})
 	if err != nil && !errors.IsNotFound(err) {
 		return err
 	}
 	if errors.IsNotFound(err) {
 		fmt.Printf("%s is not installed on this cluster\n", deploymentName)
 	} else {
-		var podsReady string
 		deploymentReady := false
-		for !deploymentReady {
+
+		if isTTY {
+			// save current cursor position
+			fmt.Print("\033[s")
+		}
+
```

Comment:
```suggestion
	tries := 0
	deploymentReady := false
```

Reply:
`tries` needs to be declared as `uint` since its compared against `timeout` which is also of type `uint` to prevent users from setting negativ timeout durations.
`uint` defaults to `0`, so we don't need to explicitly assign this value before using `tries`.